wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceRuntimes.java

```diff
@@ -245,6 +245,34 @@ public void stop(String workspaceId) throws NotFoundException, ServerException,
     }
 
     /**
+     * Adds machine into workspace runtime.
+     *
+     * @param workspaceId
+     *         id of workspace to which runtime machine should be added
+     * @param machine
+     *         machine to add to specified runtime
+     * @throws ConflictException
+     *         if given machine already exists in specified runtime
+     * @throws NotFoundException
+     *         if workspace with specified id not exists
+     */
+    public void addMachineIntoRuntime(String workspaceId, MachineImpl machine) throws ConflictException, NotFoundException {
+        final RuntimeDescriptor descriptor = descriptors.get(workspaceId);
+        if (descriptor == null) {
+            throw new NotFoundException("No runtime with \"" + workspaceId + "\" id");
```

Comment:
What is wrong with the message in the `get(...)` method?, or you want the same component to respond differently on the same use cases?